File: Knapsack01.java
package DAA;

import java.util.*;
public class Knapsack01 {
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the number of item:");
        int capacity =sc.nextInt();
        int weight[]=new int[capacity];
        int value[]=new int[capacity];
        for(int i=0;i<capacity;i++){
            System.out.println("Enter the weight and value of "+(i+1)+" element");
            weight[i]=sc.nextInt();
            value[i]=sc.nextInt();
        }
        int number=value.length;

        int k[][]=new int[number+1][capacity+1];

        for(int i=0;i<=number;i++){
            for(int j=0;j<=capacity;j++){
                if(i==0 || j==0){
                    k[i][j]=0;
                }
                else if(weight[i-1]<=j){
                    k[i][j]=Math.max(value[i-1]+k[i-1][j-weight[i-1]], k[i-1][j]);
                }
                else{
                    k[i][j]=k[i-1][j];
                }
            }
        }

        System.out.println(k[number][capacity]);
    } 
}

File: Knapsack01.cpp
#include <iostream>
using namespace std;

int main() {
    int capacity, items;
    cout << "Enter the knapsack capacity: ";
    cin >> capacity;
    cout << "Enter the number of items: ";
    cin >> items;

    int price[items + 1];
    int wt[items + 1];

    for (int i = 1; i <= items; i++) {
        cout << "Enter the price and weight for item " << i << ": ";
        cin >> price[i] >> wt[i];
    }

    int dp[items + 1][capacity + 1];

    for (int i = 0; i <= items; i++) {
        for (int j = 0; j <= capacity; j++) {
            if (i == 0 || j == 0) {
                dp[i][j] = 0;
            } else if (wt[i] <= j) {
                dp[i][j] = max(dp[i - 1][j], price[i] + dp[i - 1][j - wt[i]]);
            } else {
                dp[i][j] = dp[i - 1][j];
            }
        }
    }

    cout << "Maximum Profit Earned: " << dp[items][capacity] << "\n";
    return 0;
}